io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/writer/XLSInstanceWriter.java

```diff
@@ -125,12 +126,52 @@ else if (getContentType().getId().equals("eu.esdihumboldt.hale.io.xls.xlsx")) {
 	} // close try-iterator
 
 	/**
-	 * @param solveNestedProperties : Solve nested properties
-	 * @param selectedTypeName selected feature type
+	 * @param reporter an optional reporter
+	 * @return the types to export
+	 */
+	private Collection<? extends TypeDefinition> getTypesToExport(SimpleLog reporter) {
+		String exportTypes = getParameter(InstanceTableIOConstants.EXPORT_TYPE).as(String.class);
+
+		if (exportTypes != null) {
+			List<TypeDefinition> types = new ArrayList<TypeDefinition>();
+			String[] splitExportType = exportTypes.split(",");
+			for (String featureType : splitExportType) {
+				QName typeName = QName.valueOf(featureType);
+				TypeDefinition type = getTargetSchema().getType(typeName);
+
+				if (type == null) {
+					// try matching local name
+					type = getTargetSchema().getMappingRelevantTypes().stream()
+							.filter(t -> typeName.getLocalPart().equals(t.getName().getLocalPart()))
+							.findFirst().orElse(null);
+				}
+
+				if (type == null) {
+					// try matching display name
+					type = getTargetSchema().getMappingRelevantTypes().stream()
```

Comment:
Maybe move this common code from reader and writer class to a common utility class?